File: CapstoneProjectAPI/Models/DTOs/DepartmentResponseDto.cs
namespace CapstoneProjectAPI.Models.DTOs
{
    public class DepartmentResponseDto
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public List<DepartmentUserDto> Users { get; set; } = new();
    }

    public class DepartmentUserDto
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public bool IsAdmin { get; set; }
        public int? ManagerId { get; set; }
        public string? ManagerName { get; set; }
        public int Level { get; set; }
    }
}

File: CapstoneProjectFrontend/src/app/dtos/department.response.dto.ts
export interface DepartmentUserDto {
    id: number;
    name: string;
    email: string;
    isAdmin: boolean;
    managerId: number | null;
    managerName: string | null;
    level: number;
}

export interface DepartmentResponseDto {
    id: number;
    name: string;
    users: DepartmentUserDto[];
}